src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -107,16 +107,26 @@ private static byte HexToByte(char val)
                 return 0xFF;
         }
 
-        public static bool ContentsEqual(this byte[] a1, byte[] a2)
+        public static bool ContentsEqual(this byte[]? a1, byte[]? a2)
```

Comment:
Just to confirm - you intended to distinguish between null and empty arrays here? Null against null compares as _equal_ but null against empty compares as _not equal_?

Reply:
That's certainly what I wrote.

Really it's that

* All the parameters types show the fields as nullable, because they're also input types.
* When exported, the public fields aren't null
* This helper therefore never sees null
* But it takes many `!`s to express that.

If extended to the full "compare this parameters struct to that one", null and empty are different (though empty is "wrong"), so I went with this behavior.

Reply:
Thanks for the confirmation. :)